### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by jaescartin1**
Hi,
   I fixed some warnings that the code shows when I compile in my local installation.

Ubuntu 21.10
cmake 3.18.4
mpic++ 11.2.0
nvcc 11.5.119
pgc++ 21.9.0

### CODE REVIEW

File: domain/include/cstone/domain/domain.hpp [hunk 1]
```diff
@@ -439,14 +439,19 @@ class Domain
...
     }
 
     //! @brief return true if all array sizes are equal to value
     template<class... Arrays>
     static bool sizesAllEqualTo(std::size_t value, Arrays&... arrays)
     {
         std::array<std::size_t, sizeof...(Arrays)> sizes{arrays.size()...};
-        return std::count(begin(sizes), end(sizes), value) == sizes.size();
+
+        return (size_t)std::count(begin(sizes), end(sizes), value) == sizes.size();
```
Comment (sekelle) [inline domain/include/cstone/domain/domain.hpp]: prefer function-style cast over C-style cast:
```suggestion
        return size_t(std::count(begin(sizes), end(sizes), value)) == sizes.size();
```

File: domain/include/cstone/gravity/multipole.hpp [hunk 2]
```diff
@@ -221,7 +221,9 @@ util::tuple<T1, T1, T1, T1> particle2particle(T1 tx,
     T1 azLoc = 0;
     T1 uLoc  = 0;
 
-    #pragma clang loop vectorize(enable)
+	#if defined(__llvm__) || defined(__clang__)
+    	#pragma clang loop vectorize(enable)
+	#endif
```
Comment (sekelle) [inline domain/include/cstone/gravity/multipole.hpp]: wrong alignment, spurious tab-characters?

File: domain/include/cstone/gravity/treewalk.hpp [hunk 3]
```diff
@@ -112,7 +112,9 @@ void computeGravityGroup(TreeNodeIndex groupIdx,
             LocalParticleIndex numTargets  = lastTarget - firstTarget;
 
             // apply multipole to all particles in group
-            #pragma clang loop vectorize(enable)
+			#if defined(__llvm__) || defined(__clang__)
+				#pragma clang loop vectorize(enable)
+			#endif
```
Comment (sekelle) [inline domain/include/cstone/gravity/treewalk.hpp]: spurious tab characters? Please configure to IDE to only use space characters for indentation.

File: include/sph/findNeighborsSfc.hpp [hunk 4]
```diff
@@ -57,7 +57,7 @@ size_t neighborsSumImpl(const Task& t)
     size_t sum = 0;
 
     #pragma omp parallel for reduction(+ : sum)
-    for (unsigned int i = 0; i < t.size(); i++)
+    for (size_t i = 0; i < (size_t)t.size(); i++)
```
Comment (sekelle) [inline include/sph/findNeighborsSfc.hpp]: It would be better to change the return type to `size_t` in Task.hpp

**Review by sekelle — CHANGES_REQUESTED** [review]

File: domain/include/cstone/domain/domain.hpp (hunk 1 above)
Comment (jaescartin1) [reply domain/include/cstone/domain/domain.hpp]: Yes, sure. I worked with c-style for 2-years ... I still have automatic habits :)

File: domain/include/cstone/gravity/multipole.hpp (hunk 2 above)
Comment (jaescartin1) [reply domain/include/cstone/gravity/multipole.hpp]: You were right. I reinstalled the IDE eclipse some days ago. I forgot to change in the 'Formatted' tab by spaces. Now It is fix and commit.

File: domain/include/cstone/gravity/treewalk.hpp (hunk 3 above)
Comment (jaescartin1) [reply domain/include/cstone/gravity/treewalk.hpp]: You were right. I reinstalled the IDE eclipse some days ago. I forgot to change in the 'Formatted' tab by spaces. Now It is fix and commit.

File: include/sph/findNeighborsSfc.hpp (hunk 4 above)
Comment (jaescartin1) [reply include/sph/findNeighborsSfc.hpp]: Perfect. I removed the cast and I changed the Task.size() function. The commit is in-place.

**Review by sekelle — APPROVED** [review]

**Comment by jgphpc** [comment]
Check gitlab tests results before merging: [  FAILED  ] Ran 44/46 test case(s) from 46 check(s) (1 failure(s), 2 skipped) https://git.scicore.unibas.ch/j.piccinali/mirror_test/-/pipelines/13829 github_pr=131
0e479e5 
```
FAILURE STATISTICS

Total number of test cases: 46
Total number of failures: 1

Phase         #     Failing test cases
------------- ----- ------------------------------------------------------------
sanity        1     [Run_cascadelake_component_units_cuda, gnu+openmpi, dmi:a100]
------------------------------------------------------------------------------
Run report saved in '/users/staff/dmi-dmi/piccinal/.reframe/reports/run-report.json'
Log file(s) saved in '/users/staff/dmi-dmi/piccinal/SINGULARITY/runner/builds/AxtzSHcq/0/j.piccinali/mirror_test/tmp.13829/reframe/reframe.log', '/users/staff/dmi-dmi/piccinal/SINGULARITY/runner/builds/AxtzSHcq/0/j.piccinali/mirror_test/tmp.13829/reframe/reframe.out'
```

**Pull request merged**